pubsub/src/main/scala/org/byrde/pubsub/AdminClientTrait.scala

```diff
@@ -0,0 +1,54 @@
+package org.byrde.pubsub
+
+import com.google.api.gax.core.{CredentialsProvider, NoCredentialsProvider}
+import com.google.api.gax.grpc.GrpcTransportChannel
+import com.google.api.gax.rpc.{FixedTransportChannelProvider, TransportChannelProvider}
+import com.google.cloud.pubsub.v1.{SubscriptionAdminClient, SubscriptionAdminSettings, TopicAdminClient, TopicAdminSettings}
+import io.grpc.{ManagedChannel, ManagedChannelBuilder}
+import org.byrde.logging.Logger
+
+trait AdminClientTrait {
+  private var channel: ManagedChannel = _
+  private def transportChannel(maybeHost: Option[String])(implicit logger: Logger): Option[TransportChannelProvider] =
+    maybeHost.map(pubsubHost => {
+      if (channel != null)
```

Comment:
going with null here because is efficient and well tucked in the private method... let me know if you really do prefer that we go with Option and the unwrapping etc that comes with it... happy to put that into options if it gets this approved heh

Reply:
:+1: